diff -urN galib246/ga/gaconfig.h galib246-p/ga/gaconfig.h
--- galib246/ga/gaconfig.h	2005-01-31 16:46:09.000000000 +0100
+++ galib246-p/ga/gaconfig.h	2005-04-01 17:11:47.000000000 +0200
@@ -365,7 +365,7 @@
 #define GALIB_USE_PID
 #define GALIB_USE_EMPTY_TEMPLATES
 #define GALIB_NEED_INSTANTIATION_PREFIX
-#if __GNUC__ == 3
+#if __GNUC__ >= 3
 #define GALIB_USE_ANSI_HEADERS
 #define GALIB_USE_STD_NAMESPACE
 #define GALIB_USE_COMP_OPERATOR_TEMPLATES
diff -urN galib246/ga/gaid.h galib246-p/ga/gaid.h
--- galib246/ga/gaid.h	1999-11-11 19:56:03.000000000 +0100
+++ galib246-p/ga/gaid.h	2005-04-01 17:10:39.000000000 +0200
@@ -53,6 +53,7 @@
   int sameClass(const GAID &b) const {return(classID() == b.classID());} 
   virtual const char * className() const {return "no class";} 
   virtual int classID() const {return 0;}
+  virtual ~GAID() {}
 };
 
 
